services/user.js

```diff
@@ -26,7 +26,8 @@ const userService = {
 
   getUserById: async (id, role) => {
     return await User.findOne({ _id: id, ...(role && { role }) })
-      .populate('categories')
+      .populate({ path: 'mainSubjects.tutor', select: ['-createdAt', '-udpdatedAt'] })
+      .populate({ path: 'mainSubjects.student', select: ['-createdAt', '-udpdatedAt'] })
```

Comment:
And use an array to populate multiple paths

Reply:
Fixed